common/lib/dependabot/config/ignore_condition.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Dependabot
+  module Config
+    # Filters versions that should not be considered for dependency updates
+    class IgnoreCondition
+      UPDATE_TYPES = %i(
+        ignore_major_versions
+        ignore_minor_versions
+        ignore_patch_versions
+      ).freeze
+
+      attr_reader :dependency_name, :versions, :update_types
+      def initialize(dependency_name:, versions: nil, update_types: nil)
+        @dependency_name = dependency_name
+        @versions = versions || []
+        @update_types = update_types || []
+      end
+
+      def ignored_versions(dependency)
+        versions_by_type(dependency) + @versions
+      end
+
+      private
+
+      def versions_by_type(dep)
+        case @update_types.first # FIXME: flatmap
+        when :ignore_patch_versions
+          [ignore_version(dep.version, 4)]
+        when :ignore_minor_versions
+          [ignore_version(dep.version, 3)]
+        when :ignore_major_versions
+          [ignore_version(dep.version, 2)]
+        else
+          []
+        end
+      end
+
+      def ignore_version(version, precision)
```

Comment:
Do we want to crib some tests from find or create ignore condition in api? Looks like there's also some codes to add `.a`, assuming this is to include pre-releases. https://github.com/github/dependabot-api/blob/29bce7f4a03d42a67c5c528110b98cc7db9aec8b/app/actions/find_or_create_ignore_condition.rb#L55-L75

Reply:
That link 404s for our external friends :/
I think the equivalent `"a" for pre-release` is on L46, I'll include pre-release considerations in the test suite.